runtime/permissions/lib.rs

```diff
@@ -1101,30 +1107,65 @@ impl UnaryPermission<WriteDescriptor> {
     self.check_desc(None, false, api_name, || None)
   }
 }
-
 impl UnaryPermission<NetDescriptor> {
   pub fn query<T: AsRef<str>>(
     &self,
     host: Option<&(T, Option<u16>)>,
   ) -> PermissionState {
-    self.query_desc(
-      host.map(|h| NetDescriptor::new(&h)).as_ref(),
-      AllowPartial::TreatAsPartialGranted,
-    )
+    let net_desc_result = match host {
+      Some(&(ref host_str, port_option)) => {
+        NetDescriptor::new(&&(host_str.as_ref(), port_option))
+      }
+      None => Err(anyhow!("Host information is missing")),
+    };
+    match net_desc_result {
+      Ok(net_desc) => {
+        self.query_desc(Some(&net_desc), AllowPartial::TreatAsPartialGranted)
+      }
+
+      Err(err) => {
+        eprintln!("Error creating NetDescriptor: {}", err);
```

Comment:
Please replace all occurrences of `eprintln!` in this file with `log::error!`

Reply:
Done, lint issues it was fixed.